field-agent-ui/src/components/AddAgent.js

```diff
@@ -0,0 +1,121 @@
+    };
+
+    const middleNameOnChangeHandler = (event) => {
+        setMiddleName(event.target.value);
+    };
+
+    const lastNameOnChangeHandler = (event) => {
+        setLastName(event.target.value);
+    };
+
+    const dobOnChangeHandler = (event) => {
+        setDob(event.target.value);
+    };
+
+    const heightOnChangeHandler = (event) => {
+        setHeightInInches(event.target.value);
+    };
+
+    const addAgentFormSubmitHandler = (event) => {
+        event.preventDefault();
+
+        const newAgent = {
+            firstName,
+            middleName,
+            lastName,
+            dob,
+            heightInInches
+        };
+
+        const init = {
+            method: 'POST', // GET by default
+            headers: {
+                'Content-Type': 'application/json',
+                'Authorization': `Bearer ${auth.user.token}`
+            },
+            body: JSON.stringify(newAgent)
+        };
+
+        fetch('http://localhost:8080/api/agent', init)
+            .then(response => {
+                if (response.status === 201 || response.status === 400) {
+                    return response.json();
+                }
+                return Promise.reject('Something unexpected went wrong :)');
+            })
+            .then(data => {
+                // we either created the recorded...
+                if (data.agentId) {
+                    // redirect the user back to the /todos route
+                    history.push('/agent');
+                } else {
+                    // we have error messages
+                    setErrors(data);
+                }
+            })
+            .catch(error => console.log(error));
+    };
+
+    return (
+        <div className="container">
+            <h2 className="my-4">Add Agent</h2>
+            <Errors errors={errors} />
+            <form onSubmit={addAgentFormSubmitHandler}>
+                <div className="form-group">
+                    <label htmlFor="firstName">First Name: </label>
+                    <input className="form-control" type="text" id="firstName" name="firstName"
+                           value={firstName} onChange={firstNameOnChangeHandler} />
+                </div>
+                <div className="form-group">
+                    <label htmlFor="middleName">Middle Name: </label>
+                    <input className="form-control" type="text" id="middleName" name="middleName"
+                           value={middleName} onChange={middleNameOnChangeHandler} />
+                </div>
+                <div className="form-group">
+                    <label htmlFor="lastName">Last Name: </label>
+                    <input className="form-control" type="text" id="lastName" name="lastName"
+                           value={lastName} onChange={lastNameOnChangeHandler} />
+                </div>
+                <div className="form-group">
+                    <label htmlFor="dob">DOB: </label>
```

Comment:
You may want to use 'date' (instead of text) as the input type for dob in your forms.